_posts/2015-01-26-monster-fighter.md

```diff
@@ -0,0 +1,128 @@
+ > 700G data in the operational DB
+ > 1500 maintenance JIRA tickets updated (2014)
+ > 600 development JIRA tickets updated (2014)
+ over 50 timed batch jobs
+ some 200 integrations
+ 14 user groups
+ over 4000 users
+```
+
+These figures indicate quite a moderate complexity of the software target.  Now one interesting feature is a burden that a large system places on its developer.  I have only a limited time at my disposal, and the more complex the domain gets, the greater a share of my time that goes to domain details, versus technical and technical–professional matters.
+
+This is not a simple black/white good/bad question where domain would be icky and technology good.  Business insight and **data is often more valuable than code**, although their realisations are limited from a developer perspective.
+
+Here's a recent chart drawn from our project's data maintenance by a colleague and translated into English by me (thanks, Minna).
+
+![elmo](/img/monster-fighter/Elmo-Tietomassojen-hallinta-en.png)
+
+This is from a data maintenance perspective, on a relatively abstract level.  Code, which is our main deliverable, is excluded.  Yet already the number of interactions between components may seem daunting here.
+
+##Lesson 1: How to feel bad and ask stupid questions (sometimes twice)
+
+A big and an extensively **modularised codebase** can be hard to understand, and a complex ecosystem of professionals can be in some ways hard to relate to.  These are normal feelings and no good grounds for suicidal thoughts...  
+
+A dev on a large system needs some cold blood, but a disciplined aproach to work helps too, to counteract complexity related anxiety.  Search strategies play a central role in this puzzle.
+
+![surprise-boxes](/img/monster-fighter/surprise-boxes.png)
+
+My life is full of surprise boxes (haha)!
+
+On an ordinary workday, one might be solving a problem like this:  <font color='DarkBlue'>can I trust a source of information, or should I look for some other source to be sure</font>?  In the end, it could boil down to asking somebody's opinion, or if one remains the prima professional, reading or debugging some code, searching for documentation or incident reports, and querying a database.  Documentation is nice, but you don't generally speaking trust it unless somebody affirms that it is trustful and up to date.  **Code itself can get outdated in a large system**, since how a thing was implemented is neither the absolute truth nor should it be a *ground de facto* for business insight.
+
+Or:  <font color='DarkBlue'>what does this business concept *y* would mean and (exhaustively) what does it relate to</font>?  Now a big part of the game goes in asking, and you can tell a pro by the way they ask questions.  Maintaining a good communication channel to other parties is an art in itself.  Basically, feeling stupid is a good, constructive sign of personal human potential (unless you burnout, of course)... 
```

Comment:
"what does this business concept *y* would mean" -> "what does this business concept mean"